src/components/modals/LoginModal.tsx

```diff
@@ -22,7 +22,7 @@ export default function LoginModal({
   beforeLogin,
   ...props
 }: LoginModalProps) {
-  const login = useMyAccount((state) => state.login)
+  const { loginAnonymously, login } = useMyAccount((state) => state)
```

Comment:
this kind of pattern is bad, because if you get `(state) => state`, this component will rerender if any of the state changes

there are 2 options to fix this:
1. separate the getter to only 1 per call
```
const login = useMyAccount(state => state.login)
const loginAnonymously = useMyAccount(state => state.loginAnonymously)
```
2. use zustand shallow
```
import { shallow } from 'zustand/shallow'
const { login, loginAnonymously } = useMyAccount(state => ({
  loginAnonymously: state.loginAnonymously,
  login: state.login
}, shallow)
```

I personally prefer 1 approach especially if you only need 2 items like in this case.

Reply:
I understand that! Approach 2 seems a better case if u need more than 3-4 items